common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -144,7 +143,7 @@ trait Logging {
   implicit class LogStringContext(val sc: StringContext) {
     def log(args: MDC*): MessageWithContext = {
       val processedParts = sc.parts.iterator
-      val sb = new StringBuilder(processedParts.next())
+      val sb = new StringBuilder(StringContext.processEscapes(processedParts.next()))
```

Comment:
I think this following also needs to be dealt with:
<img width="518" alt="image" src="https://github.com/apache/spark/assets/15246973/65a1c285-3612-4187-b8ae-d6628ed5f193">

Reply:
log"This is a log message\nThis is a new line \t other msg ${MDC(LogKeys.PATHS, "C:\\Users\\run-all_1.R\nC:\\Users\\run-all_2.R")} This is a 2 log message\nThis is a 2 new line \t other 2 msg"

Reply:
Before:
```

```


After:
```
24/06/21 11:10:02 ERROR PatternLoggingSuite: This is a log message
This is a new line 	 other msg C:\Users\run-all_1.R
C:\Users\run-all_2.R This is a 2 log message
This is a 2 new line 	 other 2 msg
```

Reply:
Thanks, updated

Reply:
@gengliangwang
I have submitted the verified UT as a separate PR here.
https://github.com/apache/spark/pull/47051

Reply:
Currently, it looks very perfect.